### PR TITLE
fix(links): stop counting escaped-pipe wikilinks and non-markdown assets as broken

### PR DESCRIPTION
**Pull request by aliasunder**
## Summary

- **Escaped pipe fix**: strip trailing `\` from wikilink targets during extraction — `[[path\|alias]]` in Obsidian tables was indexed as `path\` instead of `path`, creating ~350 false broken links
- **Non-markdown asset exclusion**: skip wikilinks to recognized attachment types (images, PDFs, audio, video, canvas, csv, html, json) during link extraction — these are vault assets, not note-to-note graph edges, and inflated the broken link count by ~100
- **splitWikilink intentionally unchanged** — the note-mover uses it for link reconstruction and needs the raw `\` to preserve table syntax

## Test plan

- [x] 7 new unit tests in `links.test.ts` covering both fixes + edge cases (dots in note names, explicit `.md`, plain pipe alias)
- [x] 2 new integration tests in `search-index.test.ts` verifying `brokenLinkCount` no longer counts these false positives
- [x] Mutation test: all 8 new tests fail without the fix, pass with it
- [x] Full suite: 992 tests green, 0 regressions

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Improved handling of escaped pipe characters in wikilinks for proper target resolution.
  * Non-note assets (images, PDFs, audio, video, etc.) are now correctly excluded from link processing.
  * Enhanced broken link detection to avoid flagging asset references as broken.

* **Tests**
  * Added comprehensive test coverage for escaped pipe handling and asset filtering.

### CODE REVIEW

**Comment by aliasunder** [comment]
Re: qodo finding on `isNonMarkdownTarget` dropping `[[report.pdf]]` before resolution —

This is a false positive. In Obsidian, `[[report.pdf]]` resolves to the PDF file itself — Obsidian only auto-appends `.md` when the target lacks a recognized file extension. A note named `report.pdf.md` cannot be reached via `[[report.pdf]]` in Obsidian; the user would need to write `[[report.pdf.md]]` (which passes the filter since `.md` is not in `NON_MARKDOWN_EXTENSIONS`).

The extraction-time filter correctly mirrors Obsidian's own resolution behavior: wikilinks to recognized attachment types are asset references, not note links. Moving the filter to the index layer would leak a parser-level concern into the wrong module.

**Comment by aliasunder** [comment]
@coderabbit review

File: src/vault-mcp/search/__tests__/search-index.test.ts [hunk 1]
```diff
@@ -2074,6 +2074,38 @@ describe("brokenLinkCount", () => {
...
+    index.upsertNote(
+      {
+        filePath: "source.md",
+        rawContent: "# Source\n\n![[photo.png]] and [[report.pdf]].\n",
+        fileStat: testStat(1000),
+      },
+      logger,
+    )
+    expect(index.brokenLinkCount({}, logger)).toBe(0)
+  })
```
[the remark this answers is not in the document]
Comment (aliasunder) [reply src/vault-mcp/search/__tests__/search-index.test.ts:2116]: Valid finding — strengthened both tests in a50e01a:

- **Escaped-pipe test**: now asserts the outgoing link exists and resolved (`exists: true`) before checking `brokenLinkCount === 0`
- **Non-note assets test**: added a real note link (`[[real-note]]`) as a control, asserts it appears in outgoing links and is the only broken link (`brokenLinkCount === 1`) — proving extraction ran and asset links were genuinely excluded, not silently dropped
